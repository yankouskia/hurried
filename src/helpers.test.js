const {
  createRequestMessage,
  createResponseMessage,
  isRequestMessage,
  isResponseMessage,
} = require('./helpers');

describe('helpers', () => {
  const randomFn = Math.random();

  beforeEach(() => {
    Math.random = jest.fn(() => 'id');
  });

  afterEach(() => {
    Math.random = randomFn;
  });

  describe('createRequestMessage', () => {
    it('one argument', () => {
      expect(createRequestMessage('functionName', ['param1'])).toMatchSnapshot();
    });

    it('two arguments', () => {
      expect(createRequestMessage('functionName', ['param1', {}])).toMatchSnapshot();
    });
  });

  describe('createResponseMessage', () => {
    it('empty', () => {
      expect(createResponseMessage('functionName', 'id')).toMatchSnapshot();
    });

    it('with data', () => {
      expect(createResponseMessage('functionName', 'id', 'data')).toMatchSnapshot();
    });

    it('with error', () => {
      expect(createResponseMessage('functionName', 'id', null, 'error message')).toMatchSnapshot();
    });
  });

  describe('isRequestMessage', () => {
    it('true', () => {
      expect(isRequestMessage({ functionName: 'fn', id: 'id', params: [] })).toBeTruthy();
    });

    it('false', () => {
      expect(isRequestMessage('smth wrong')).toBeFalsy();
    });
  });

  describe('isResponseMessage', () => {
    it('error', () => {
      expect(isResponseMessage({ functionName: 'fn', id: 'id', error: 'error occured' })).toBeTruthy();
    });

    it('data', () => {
      expect(isResponseMessage({ functionName: 'fn', id: 'id', data: 'result' })).toBeTruthy();
    });

    it('without data and error', () => {
      expect(isResponseMessage({ functionName: 'fn', id: 'id' })).toBeTruthy();
    });

    it('false', () => {
      expect(isResponseMessage('smth wrong')).toBeFalsy();
    });
  });
});

