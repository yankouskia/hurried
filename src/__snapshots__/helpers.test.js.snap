// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers createRequestMessage one argument 1`] = `
Object {
  "functionName": "functionName",
  "id": "id",
  "params": Array [
    "param1",
  ],
}
`;

exports[`helpers createRequestMessage two arguments 1`] = `
Object {
  "functionName": "functionName",
  "id": "id",
  "params": Array [
    "param1",
    Object {},
  ],
}
`;

exports[`helpers createResponseMessage empty 1`] = `
Object {
  "data": undefined,
  "error": undefined,
  "functionName": "functionName",
  "id": "id",
}
`;

exports[`helpers createResponseMessage with data 1`] = `
Object {
  "data": "data",
  "error": undefined,
  "functionName": "functionName",
  "id": "id",
}
`;

exports[`helpers createResponseMessage with error 1`] = `
Object {
  "data": null,
  "error": "error message",
  "functionName": "functionName",
  "id": "id",
}
`;
